app/src/main/java/com/imdbmovieapp/data/local/dao/MovieDao.kt

```diff
@@ -0,0 +1,22 @@
+package com.imdbmovieapp.data.local.dao
+
+import androidx.lifecycle.LiveData
+import androidx.room.Dao
+import androidx.room.Delete
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.imdbmovieapp.data.local.entity.MovieEntity
+
+@Dao
+interface MovieDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(movieEntity: MovieEntity)
+
+    @Delete
+    suspend fun delete(movieEntity: MovieEntity)
+
+    @Query
+        ("SELECT * FROM 'movieTable'")
+    fun getAllMovies(): LiveData<List<MovieEntity>>
```

Comment:
Flow is more flexible and integrates better with coroutines. It also has better support for operators (e.g., mapping, filtering), and Room optimizes its queries to work seamlessly with Flow.